### PR TITLE
refactor(*): merge all EvictionConfiguration fields to AdminQoSConfiguration

### PR DESCRIPTION
**Pull request by luomingmeng**
The eviction configuration also is part of AdminQoSConfiguration, so we merge them into a same crd.

### CODE REVIEW

File: pkg/apis/config/v1alpha1/adminqos.go [hunk 1]
```diff
@@ -62,7 +62,12 @@ type AdminQoSConfigurationSpec struct {
 
 type AdminQoSConfig struct {
 	// ReclaimedResourceConfig is a configuration for reclaim resource
-	ReclaimedResourceConfig ReclaimedResourceConfig `json:"reclaimedResourceConfig,omitempty"`
+	// +optional
+	ReclaimedResourceConfig *ReclaimedResourceConfig `json:"reclaimedResourceConfig,omitempty"`
+
+	// EvictionConfig is a configuration for eviction
+	// +optional
+	EvictionConfig *EvictionConfig `json:"evictionConfiguration"`
```
Comment (caohe) [inline pkg/apis/config/v1alpha1/adminqos.go:70]: Would the thresholds of `drop cache` and `tune memcg` be included in this field?

Comment (luomingmeng) [reply pkg/apis/config/v1alpha1/adminqos.go:70]: They may be included in another field such as `MemorySupressConfig` in the future, and `EvictionConfig` will only include the field that about pod eviction.

File: pkg/apis/config/v1alpha1/adminqos.go [hunk 2]
```diff
@@ -88,4 +93,207 @@ type ReclaimedResourceConfig struct {
...
+}
+
+type EvictionConfig struct {
+	// EvictionManagerConfig is the config for eviction manager
+	// +optional
+	EvictionManagerConfig *EvictionManagerConfig `json:"evictionManagerConfig,omitempty"`
+
+	// EvictionPluginsConfig is the config for all eviction plugins
+	// +optional
+	EvictionPluginsConfig *EvictionPluginsConfig `json:"evictionPluginsConfig,omitempty"`
```
Comment (caohe) [inline pkg/apis/config/v1alpha1/adminqos.go:171]: Do we need to expose implementation-related concepts such as manager and plugin to users?

**Review by csfldf — DISMISSED** [review]

**Review by caohe — DISMISSED** [review]
LGTM

**Review by sun-yuliang — APPROVED** [review]

**Review by sun-yuliang — APPROVED** [review]

**Review by csfldf — APPROVED** [review]

**Review by waynepeking348 — APPROVED** [review]

**Review by sun-yuliang — APPROVED** [review]

**Pull request merged**